src/ocean/Assets.ts

```diff
@@ -638,4 +643,30 @@ export class Assets extends Instantiable {
     }
     return results
   }
+
+  /**
+   *
+   * @param {String} did
+   * @return {Promise<Consumable>}
+   */
+  public async isConsumable(did: string): Promise<Consumable> {
+    const ddo = await this.resolve(did)
```

Comment:
from the client side, for the use case for disabling the buy button when user arrives at asset details page, it would help to not fetch the DDO here again cause in almost all use case we already have the DDO. So should either accept a DDO or DID. And when DDO is passed, it should not refetch the DDO.